shaka/src/js/dom/attr.cc

```diff
@@ -21,15 +21,27 @@ namespace shaka {
 namespace js {
 namespace dom {
 
-Attr::Attr(RefPtr<Element> owner, const std::string& local_name,
+Attr::Attr(RefPtr<Element> owner,
+           const std::string& local_name,
            optional<std::string> namespace_uri,
-           optional<std::string> namespace_prefix, const std::string& value)
-    : Node(ATTRIBUTE_NODE, owner->document()),
-      namespace_uri(namespace_uri),
+           optional<std::string> namespace_prefix,
+           const std::string& value)
+    : Attr(owner->document(), owner, local_name, namespace_uri, namespace_prefix, value) {}
+
+Attr::Attr(RefPtr<Document> document,
+           RefPtr<Element> owner,
+           const std::string& local_name,
+           optional<std::string> namespace_uri,
+           optional<std::string> namespace_prefix,
+           const std::string& value)
+    : Node(ATTRIBUTE_NODE, document),
+      namespace_uri(namespace_uri.has_value() && !namespace_uri.value.empty() ? namespace_uri : nullopt),
```

Comment:
This seems like something the caller should do.  If I try `document.createAttribute('', 'a:foo')` in Chrome, it throws an error.  So the caller should check this.

Reply:
That's because `createAttribute` only takes one argument. Did you mean `createAttributeNS`? In any case, this is configured this way because empty namespace URIs when using `createAttributeNS` ultimately map this value to `null`.